mldatafind/find.py

```diff
@@ -1,32 +1,36 @@
-from concurrent.futures import (
-    FIRST_COMPLETED,
-    ProcessPoolExecutor,
-    ThreadPoolExecutor,
-    wait,
-)
+import logging
+import os
+from concurrent.futures import ProcessPoolExecutor, ThreadPoolExecutor
 from functools import partial
 from pathlib import Path
-from typing import Callable, Iterable, Iterator, List, Optional
+from typing import TYPE_CHECKING, Callable, Iterable, Iterator, List, Optional
+
+import numpy as np
+
+if TYPE_CHECKING:
+    from concurrent.futures import Future
 
 from gwpy.segments import Segment, SegmentList
 
+import mldatafind.utils as utils
 from mldatafind.io import fetch_timeseries, read_timeseries
 from mldatafind.segments import query_segments
 
-MEMORY_LIMIT = 1e10  # ? in bytes
-BITS_PER_BYTE = 8
+DEFAULT_SEGMENT_SERVER = os.getenv(
+    "DEFAULT_SEGMENT_SERVER", "https://segments.ligo.org"
+)
 
+MEMORY_LIMIT = 5  # GB
 
-def _calc_memory(
-    n_channels: int,
-    duration: float,
-    precision: int = 64,
-    sample_rate: float = 16384.0,
-):
 
-    n_samples = n_channels * duration * sample_rate
-    num_bytes = n_samples * (precision / BITS_PER_BYTE)
-    return num_bytes
+def _handle_future(future: "Future"):
```

Comment:
So I've realized that this is unnecessary when I included it in BBHNet: calling `future.result()` will raise any exception that got thrown in the target

Reply:
And will actually do it better by including the traceback, which this method actually throws away